tests/components/russound_rio/test_config_flow.py

```diff
@@ -89,6 +108,117 @@ async def test_duplicate(
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {},
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "MCA-C5"
+    assert result["data"] == {
+        CONF_HOST: "192.168.20.17",
+        CONF_PORT: 9621,
+    }
+    assert result["result"].unique_id == "00:11:22:33:44:55"
+
+
+async def test_zeroconf_flow_errors(
+    hass: HomeAssistant,
+    mock_russound_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test zeroconf flow."""
+    mock_russound_client.connect.side_effect = TimeoutError
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=ZEROCONF_DISCOVERY,
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
+
+    mock_russound_client.connect.side_effect = None
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=ZEROCONF_DISCOVERY,
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "discovery_confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {},
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "MCA-C5"
+    assert result["data"] == {
+        CONF_HOST: "192.168.20.17",
+        CONF_PORT: 9621,
+    }
+    assert result["result"].unique_id == "00:11:22:33:44:55"
+
+
+async def test_zeroconf_duplicate(
+    hass: HomeAssistant,
+    mock_russound_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test duplicate flow."""
+    mock_config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=ZEROCONF_DISCOVERY,
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_zeroconf_duplicate_different_ip(
+    hass: HomeAssistant,
+    mock_russound_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test duplicate flow with different IP."""
+    mock_config_entry.add_to_hass(hass)
+
+    ZEROCONF_DISCOVERY.ip_address = (ip_address("192.168.20.18"),)
+    ZEROCONF_DISCOVERY.ip_addresses = [ip_address("192.168.20.18")]
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=ZEROCONF_DISCOVERY,
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+    entry = hass.config_entries.async_get_entry(mock_config_entry.entry_id)
+    assert entry
+    assert entry.data == {
+        CONF_HOST: "(IPv4Address('192.168.20.18'),)",
```

Comment:
Can't figure out why this is what is being returned.. The value is updated from discovery_info.host, so it should just be 192.168.20.18 ...